app/src/main/java/com/randomappsinc/studentpicker/presentation/PresentationAdapter.kt

```diff
@@ -0,0 +1,87 @@
+package com.randomappsinc.studentpicker.presentation
+
+import android.graphics.Color
+import android.text.TextUtils
+import android.util.TypedValue
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ImageView
+import android.widget.TextView
+import androidx.recyclerview.widget.RecyclerView
+import com.randomappsinc.studentpicker.databinding.ChooseNameDialogCellBinding
+import com.randomappsinc.studentpicker.models.NameDO
+import com.randomappsinc.studentpicker.utils.NameUtils
+import com.squareup.picasso.Picasso
+
+class PresentationAdapter : RecyclerView.Adapter<PresentationAdapter.ViewHolder>() {
+
+    var chosenNames = ArrayList<NameDO>()
+    var showAsList = false
+
+    fun setChosenNames(chosenNames: List<NameDO>, showAsList: Boolean) {
+        this.showAsList = showAsList
+        this.chosenNames.clear()
+        this.chosenNames.addAll(chosenNames)
+        notifyDataSetChanged()
+    }
+
+    var textSize = 27f
+        set(value) {
+            field = value
+            notifyDataSetChanged()
+        }
+
+    var textColor = Color.GRAY
+        set(value) {
+            field = value
+            notifyDataSetChanged()
+        }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
+        return ViewHolder(
+                ChooseNameDialogCellBinding.inflate(
+                        LayoutInflater.from(parent.context), parent, false))
+    }
+
+    override fun getItemCount(): Int {
+        return chosenNames.size
+    }
+
+    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
+        holder.loadName(position)
+    }
+
+    inner class ViewHolder(itemBinding: ChooseNameDialogCellBinding) : RecyclerView.ViewHolder(itemBinding.root) {
+
+        private val nameNumber: TextView = itemBinding.personNumber
+        private val personImageView: ImageView = itemBinding.personImage
+        private val name: TextView = itemBinding.personName
+
+        fun loadName(position: Int) {
+            val nameDO: NameDO = chosenNames[position]
+            if (showAsList) {
```

Comment:
The vast majority of users won't have showAsList setting (it's a pretty niche setting). This adds more code complexity, but let's change the gravity of the container based on this setting to adhere to the old implementation. So if this setting is on, we keep the regular align start since we need to align the numbers. If it's off, set the gravity of the content inside to be centered horizontally. It looks really weird to only select 1 thing and have it off to the side.

Reply:
yes, it looks very weird thanks for the feedback